### DISCUSSION TITLE
flashlight/app/asr/tutorial/InferenceCTC.cpp - Code query

### DISCUSSION THREAD

**Discussion by TzurV** [flashlight/flashlight#731] [Q&A] [answered]
**Is the Trie created includes just the Unigram and a single word pronunciation that is extracted from the lexicon ?** 

flashlight/app/asr/tutorial/InferenceCTC.cpp: line 167

      std::shared_ptr<fl::lib::text::Trie> trie = fl::app::asr::buildTrie(
          "wrd" /* decoderType */,
          true /* useLexicon */,
          lm,
          "max" /* smearing */,
          tokenDict,
          lexicon,
          wordDict,
          wordSepIdx,
          0 /* repLabel */);
      LOG(INFO) << "[Inference tutorial for CTC] Trie is planted.";

I'm probably reading it wrongly, but I can't see were the NGram is used.

**Comment by tlikhomanenko** [comment, marked as answer]
Yep, it is building only unigram Trie. The ngram LM is used inside decoder itself when the leaf of trie is reached. See for example https://github.com/flashlight/flashlight/blob/master/flashlight/lib/text/decoder/LexiconDecoder.cpp#L124-L127
